compose/cli/log_printer.py

```diff
@@ -19,18 +19,21 @@
 
 class LogPresenter(object):
 
-    def __init__(self, prefix_width, color_func):
+    def __init__(self, prefix_width, color_func, config=None):
         self.prefix_width = prefix_width
         self.color_func = color_func
+        self.prefixes = {s['name']: s['container_name'] for s in filter(
+            lambda service: service.get('container_name', False), config.services if config else [])}
 
     def present(self, container, line):
-        prefix = container.name_without_project.ljust(self.prefix_width)
+        prefix = self.prefixes.get(
+            container.service, container.name_without_project).ljust(self.prefix_width)
         return '{prefix} {line}'.format(
             prefix=self.color_func(prefix + ' |'),
             line=line)
 
 
-def build_log_presenters(service_names, monochrome):
+def build_log_presenters(service_names, monochrome, config=None):
```

Comment:
The `prefix_width` can no longer be calculated just from the `service_names` as the longest name could now be a `container_name`.

As part of this change, we may be able to avoid passing the whole `config` to `__init__`.